src/Adapter/Product/CommandHandler/AddCustomizationFieldHandler.php

```diff
@@ -28,56 +28,76 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
 
-use CustomizationField as CustomizationFieldEntity;
+use CustomizationField;
 use PrestaShop\PrestaShop\Adapter\Product\AbstractCustomizationFieldHandler;
+use PrestaShop\PrestaShop\Adapter\Product\CustomizationFieldPersister;
+use PrestaShop\PrestaShop\Adapter\Product\CustomizationFieldValidator;
+use PrestaShop\PrestaShop\Adapter\Product\ProductProvider;
+use PrestaShop\PrestaShop\Adapter\Product\ProductUpdater;
 use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Command\AddCustomizationFieldCommand;
 use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\AddCustomizationFieldHandlerInterface;
-use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Exception\CannotAddCustomizationFieldException;
-use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Exception\CustomizationException;
 use PrestaShop\PrestaShop\Core\Domain\Product\Customization\ValueObject\CustomizationFieldId;
-use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
-use PrestaShopException;
 
 /**
- * Handles @var AddCustomizationFieldCommand using legacy object model
+ * Handles @see AddCustomizationFieldCommand using legacy object model
  */
 final class AddCustomizationFieldHandler extends AbstractCustomizationFieldHandler implements AddCustomizationFieldHandlerInterface
 {
+    /**
+     * @var ProductProvider
+     */
+    private $productProvider;
+
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @var CustomizationFieldPersister
+     */
+    private $customizationFieldPersister;
+
+    /**
+     * @var CustomizationFieldValidator
+     */
+    private $customizationFieldValidator;
+
+    /**
+     * @param ProductProvider $productProvider
+     * @param ProductUpdater $productUpdater
+     * @param CustomizationFieldPersister $customizationFieldPersister
+     * @param CustomizationFieldValidator $customizationFieldValidator
+     */
+    public function __construct(
+        ProductProvider $productProvider,
+        ProductUpdater $productUpdater,
+        CustomizationFieldPersister $customizationFieldPersister,
+        CustomizationFieldValidator $customizationFieldValidator
+    ) {
+        $this->productProvider = $productProvider;
+        $this->customizationFieldPersister = $customizationFieldPersister;
+        $this->customizationFieldValidator = $customizationFieldValidator;
```

Comment:
You don't need to inject this one anymore, since it's already injected and managed by the persister